### PR TITLE
[EZQMS-287] Added generic menu items and collaborator utils

### PR DESCRIPTION
**Pull request by sol-0**
# Contribution checklist

## Brief description

https://front.hc.engineering/workbench/platform/tracker/EZQMS-287

1. Added collaborator action to copy document fields
2. Refactored collaborator operations to use them from other plugins

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [x] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [x] - Does the translations are up to date?
* [x] - Does it well tested?
* [x] - Tested for Chrome.
* [ ] - Tested for Safari.
* [x] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [x] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [x] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

### CODE REVIEW

File: packages/text-editor/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  }
+  | {
+    collaboratorURL: string
+    token: string
+  }
+) & { ydoc?: Y.Doc }
+
+function getProvider (documentId: string, providerData: ProviderData, initialContentId?: string): TiptapCollabProvider {
+  if (!('provider' in providerData)) {
+    return new TiptapCollabProvider({
```
Comment (aonnikov) [inline packages/text-editor/src/utils.ts]: Do we need to destroy the created provider?

File: packages/ui/src/components/Menu.svelte [hunk 2]
```diff
@@ -14,16 +14,17 @@
...
-  import { Action } from '../types'
+  import { Action, MenuItem, MenuComponent } from '../types'
   import Icon from './Icon.svelte'
   import Label from './Label.svelte'
   import MouseSpeedTracker from './MouseSpeedTracker.svelte'
   import { resizeObserver } from '../resize'
+  import Component from './Component.svelte'
 
-  export let actions: Action[] = []
+  export let items: MenuItem[] = []
```
Comment (aonnikov) [inline packages/ui/src/components/Menu.svelte]: @haiodo please take a look at these changes

Comment (aonnikov) [reply packages/ui/src/components/Menu.svelte]: Per my point of view, we should keep menus simple and do not allow to render complex components inside, at least at platform level. Maybe you can add custom menu that allows this.

But as a user I don't understand how to use this component, I need to click into this small window, enter a value and then confirm the input somehow... I think that a popup window would be more convenient for this.

Comment (lexiv0re) [reply packages/ui/src/components/Menu.svelte]: @aonnikov let's discuss this component with @AlexandrLapin on UX call tomorrow

File: packages/text-editor/src/utils.ts (hunk 1 above)
Comment (sol-0) [reply packages/text-editor/src/utils.ts]: Fixed, thanks. I haven't used new `Disposable` TS 5.2 feature as it's not working flawlessly out-of-the-box.

File: packages/ui/src/components/Menu.svelte (hunk 2 above)
Comment (sol-0) [reply packages/ui/src/components/Menu.svelte]: I've reverted the changes. Thanks.

**Review by aonnikov — APPROVED** [review]

**Review by lexiv0re — APPROVED** [review]
@sol-0 please check the ui tests

**Pull request merged**